### PR TITLE
cluserter merger refine

### PR DESCRIPTION
**Pull request by HetaoWang**

### CODE REVIEW

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/merger/MergerFactory.java [hunk 1]
```diff
@@ -49,34 +56,51 @@ public void setScopeModel(ScopeModel scopeModel) {
             throw new IllegalArgumentException("returnType is null");
         }
 
-        Merger result;
-        if (returnType.isArray()) {
-            Class type = returnType.getComponentType();
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/merger/MergerFactory.java:54]: What is the purpose of removing `returnType.getComponentType()`?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/merger/MergerFactory.java (hunk 1 above)
Comment (HetaoWang) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/merger/MergerFactory.java:54]: one question is why the parameter type of Merger.merge() must be array? If use returnType.getComponentType() in MergerFactory, other non-array types can not exist in MergerFactory. 
for example int[] and int type can not both exist in MergerFactory.

Comment (HetaoWang) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/merger/MergerFactory.java:54]: > Another problem is MergerFactory caches merger by return type. But there may exist more than one merger with same return type, for example IntFindAnyMerger and IntFindFirstMerger in this pr.

**Comment by chickenlj** [comment]
As mentioned above, please provide the purposes of this PR. What problems have you been facing and how does this patch can solve.

**Pull request merged**